resources/app.rb

```diff
@@ -60,55 +62,61 @@
 end
 
 action :add do
-  if current_resource.path.empty?
+  if exists
+    Chef::Log.debug("#{new_resource.inspect} app already exists - nothing to do")
+  else
     converge_by "Creating the Application - \"#{new_resource}\"" do
       cmd = "#{appcmd(node)} add app /site.name:\"#{new_resource.site_name}\""
       cmd << " /path:\"#{new_resource.path}\""
       cmd << " /applicationPool:\"#{new_resource.application_pool}\"" if new_resource.application_pool
-      cmd << " /physicalPath:\"#{windows_cleanpath(new_resource.physical_path)}\"" if new_resource.physical_path
+      cmd << " /physicalPath:\"#{new_resource.physical_path}\"" if new_resource.physical_path
       cmd << " /enabledProtocols:\"#{new_resource.enabled_protocols}\"" if new_resource.enabled_protocols
       cmd << ' /commit:\"MACHINE/WEBROOT/APPHOST\"'
       Chef::Log.debug(cmd)
       shell_out!(cmd)
     end
-  else
-    Chef::Log.debug("#{new_resource.inspect} app already exists - nothing to do")
   end
 end
 
 action :config do
-  # only get the beginning of the command if there is something that changes
-  cmd = cmd_set_app
-  converge_if_changed :path do
-    # adds path to the cmd
-    cmd << " /path:\"#{new_resource.path}\"" if new_resource.path
-  end
-  converge_if_changed :application_pool do
-    # adds applicationPool to the cmd
-    cmd << " /applicationPool:\"#{new_resource.application_pool}\"" if new_resource.application_pool
-  end
-  converge_if_changed :enabled_protocols do
-    # adds enabledProtocols to the cmd
-    cmd << " /enabledProtocols:\"#{new_resource.enabled_protocols}\"" if new_resource.enabled_protocols
-  end
-  Chef::Log.debug(cmd)
+  if exists
+    # only get the beginning of the command if there is something that changes
+    cmd = cmd_set_app
+    converge_if_changed :path do
+      # adds path to the cmd
+      cmd << " /path:\"#{new_resource.path}\"" if new_resource.path
+    end
+    converge_if_changed :application_pool do
+      # adds applicationPool to the cmd
+      cmd << " /applicationPool:\"#{new_resource.application_pool}\"" if new_resource.application_pool
+    end
+    converge_if_changed :enabled_protocols do
+      # adds enabledProtocols to the cmd
+      cmd << " /enabledProtocols:\"#{new_resource.enabled_protocols}\"" if new_resource.enabled_protocols
+    end
+    Chef::Log.debug(cmd)
 
-  if cmd == cmd_set_app
-    Chef::Log.debug("#{new_resource.inspect} application - nothing to do")
-  else
-    shell_out!(cmd)
-  end
+    if cmd == cmd_set_app
+      Chef::Log.debug("#{new_resource.inspect} application - nothing to do")
+    else
+      converge_by "Updating the Application - \"#{new_resource}\"" do
```

Comment:
Just a friendly message wrapper, and will keep the command from executing (which it shouldn't anyways since all above commands are wrapped with `converge_if_changed`